jupyterhub/orm.py

```diff
@@ -1067,6 +1048,8 @@ def new_session_factory(
     elif url.startswith('mysql'):
         kwargs.setdefault('pool_recycle', 60)
 
+    kwargs.setdefault("future", True)
```

Comment:
What are the circumstances where a caller should be allowed to pass `future=False`?

Reply:
Good question! If someone were using sqlalchemy 1.4 and found a _problem_ with our switch to 2.0 lifecycles, this gives them an escape hatch to keep the older behavior without needing to downgrade to JupyterHub. It should have no practical effect if everything's been done correctly. The alternatives are:

- unconditional `kwargs['future'] = True` (always get 2.0 behavior, even on 1.4, no escape hatch but always consistent hardcoded behavior)
- leave it out (status quo - get deprecated but default 1.x behavior on 1.4, 2.0 behavior on upgrade or user opt-in)

This is a no-op on 2.0.

Reply:
Thanks, I'll let you decide!